### DISCUSSION TITLE
NEAT Algorithm Support?

### DISCUSSION THREAD

**Discussion by ubeydtalha** [anyoptimization/pymoo#260] [Ideas]
Will the NEAT algorithm be added to this library in future versions? It has a coded library for Python, but its interface is not as nice and useful as this library. I know the NEAT algorithm is for establishing a neural network, but I think it would be legendary if it was in this library.

Paper: http://nn.cs.utexas.edu/downloads/papers/stanley.ec02.pdf

**Comment by blankjul** [comment]
The algorithm looks interesting and I have heard of it before. Is the Python implementation the original one implemented by the authors of the paper? Is it somewhere publicly available?

**Reply by ubeydtalha** [reply]
There is a Github repository, but it hasn't been developed for a long time.
I don't think it belongs to the authors of the paper.

https://github.com/CodeReclaimers/neat-python

Also , there is NSGAII implemented version :
https://github.com/hugoaboud/neat-python/tree/master/neat/nsga2

**Comment by blankjul** [comment]
I had a look at it, and - correct me if I am wrong - NEAT is directly designed for optimizing neural networks. Current implementations in pymoo generalize to at least a category of problems.

However, please let me know if you end up implementing it in a fork of the pymoo framework. I am for sure willing to have a look!

**Reply by ubeydtalha** [reply]
same to train more networks, but while training the network, optimization is done at the same time by using genetic algorithms. Thus, both the optimum solution and an artificial neural network for the problem are established in your hands. My suggestion was that while establishing the network with this algorithm, it should also be optimized with the desired genetic algorithm in the pymoo library. I'm a bit new to these issues and the python library of the NEAT algorithm was very confusing and I couldn't find a proper explanation. That's why I appreciate Pymoo. I'll try using NEAT and Pymoo together but I'm not sure I can :)

**Reply by gsause** [reply]
Did you try doing that?

**Reply by ubeydtalha** [reply]
> Did you try doing that?
> 

Unfortunately, I did not have time to try. 
But , I have an idea for creating ai with using genetic algorithms.
